src/ansys/mechanical/core/embedding/app.py

```diff
@@ -330,3 +330,78 @@ def update_globals(
     def _update_all_globals(self) -> None:
         for scope in self._updated_scopes:
             scope.update(global_entry_points(self))
+
+    def _print_tree(self, node, max_lines, lines_count, indentation):
+        """Recursively print till provided maximum lines limit.
+
+        Parameters
+        ----------
+        lines_count: int, optional
+            The current count of lines printed. Default is 0.
+        indentation: str, optional
+            The indentation string used for printing the tree structure. Default is "".
+        """
+        if lines_count >= max_lines and max_lines != -1:
+            print(f"... truncating after {max_lines} lines")
+            return lines_count
+
+        if not hasattr(node, "Name"):
+            raise AttributeError("Object must have a 'Name' attribute")
+
+        node_name = node.Name
+        if hasattr(node, "Suppressed") and node.Suppressed is True:
+            node_name += " (Suppressed)"
+        print(f"{indentation}├── {node.Name}")
+        lines_count += 1
+
+        if hasattr(node, "Children") and node.Children is not None and node.Children.Count > 0:
+            for child in node.Children:
+                _lines_count = self._print_tree(child, max_lines, lines_count, indentation + "|  ")
+                if _lines_count >= max_lines and max_lines != -1:
+                    break
+
+        return lines_count
+
+    def print_tree(self, node, max_lines=80, lines_count=0, indentation=""):
+        """
+        Print the hierarchical tree representation of the Mechanical project structure.
+
+        Each object in the tree is expected to have the following attributes:
```

Comment:
The paragraph about "Each object" doesn't belong  here, its not relevant to the user. It can be in the docstring of the internal method